<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd"><html xmlns="http://www.w3.org/1999/xhtml" lang="ru"><head><meta http-equiv="Content-Type" content="text/html;charset=UTF-8"/><link rel="stylesheet" href="../jacoco-resources/report.css" type="text/css"/><link rel="shortcut icon" href="../jacoco-resources/report.gif" type="image/gif"/><title>ExceptionsHandler.java</title><link rel="stylesheet" href="../jacoco-resources/prettify.css" type="text/css"/><script type="text/javascript" src="../jacoco-resources/prettify.js"></script></head><body onload="window['PR_TAB_WIDTH']=4;prettyPrint()"><div class="breadcrumb" id="breadcrumb"><span class="info"><a href="../jacoco-sessions.html" class="el_session">Sessions</a></span><a href="../index.html" class="el_report">CloudStorage</a> &gt; <a href="index.source.html" class="el_package">com.visage.cloudstorage.Exceptions</a> &gt; <span class="el_source">ExceptionsHandler.java</span></div><h1>ExceptionsHandler.java</h1><pre class="source lang-java linenums">package com.visage.cloudstorage.Exceptions;

import com.visage.cloudstorage.Model.ErrorResponse;
import org.springframework.dao.DataIntegrityViolationException;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.security.authentication.BadCredentialsException;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
<span class="fc" id="L12">public class ExceptionsHandler {</span>

    @ExceptionHandler(Exception.class)
    public ResponseEntity&lt;?&gt; handleException(Exception e){
<span class="nc" id="L16">        return ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(ErrorResponse.builder()</span>
<span class="nc" id="L17">                .status(500)</span>
<span class="nc" id="L18">                .message(&quot;Неизвестная ошибка&quot;)</span>
<span class="nc" id="L19">                .build());</span>
    }

    @ExceptionHandler(RuntimeException.class)
    public ResponseEntity&lt;?&gt; runtime(){
<span class="nc" id="L24">        return ResponseEntity.status(HttpStatus.CONFLICT).body(ErrorResponse.builder()</span>
<span class="nc" id="L25">                .status(500)</span>
<span class="nc" id="L26">                .message(&quot;Неизвестная ошибка&quot;)</span>
<span class="nc" id="L27">                .build()</span>
        );
    }

    @ExceptionHandler(DataIntegrityViolationException.class)
    public ResponseEntity&lt;?&gt; userAlreadyExists(){
<span class="nc" id="L33">        return ResponseEntity.status(HttpStatus.CONFLICT).body(ErrorResponse.builder()</span>
<span class="nc" id="L34">                .status(409)</span>
<span class="nc" id="L35">                .message(&quot;Пользователь с таким именем уже существует&quot;)</span>
<span class="nc" id="L36">                .build()</span>
        );
    }

    @ExceptionHandler(BadCredentialsException.class)
    public ResponseEntity&lt;?&gt; badUsernameOrPassword(){
<span class="fc" id="L42">        return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(ErrorResponse.builder()</span>
<span class="fc" id="L43">                .status(401)</span>
<span class="fc" id="L44">                .message(&quot;Введены неверные данные&quot;)</span>
<span class="fc" id="L45">                .build());</span>
    }


    @ExceptionHandler(DataNotFoundException.class)
    public ResponseEntity&lt;?&gt; dataNotFoundException(DataNotFoundException e){
<span class="fc" id="L51">        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ErrorResponse.builder()</span>
<span class="fc" id="L52">                .status(404)</span>
<span class="fc" id="L53">                .message(e.getMessage())</span>
<span class="fc" id="L54">                .build());</span>
    }

    @ExceptionHandler(NotCorrectNameFileOrPackage.class)
    public ResponseEntity&lt;?&gt; notCorrectNameFileOrPackage(){
<span class="nc" id="L59">        return ResponseEntity.status(HttpStatus.CONFLICT).body(ErrorResponse.builder()</span>
<span class="nc" id="L60">                .status(409)</span>
<span class="nc" id="L61">                .message(&quot;Такое имя файла уже существует&quot;)</span>
<span class="nc" id="L62">                .build());</span>

    }

    @ExceptionHandler(ParentFileIsAlreadyExists.class)
    public ResponseEntity&lt;?&gt; parentNotExists(){
<span class="nc" id="L68">        return ResponseEntity.status(HttpStatus.NOT_FOUND).body(ErrorResponse.builder()</span>
<span class="nc" id="L69">                .status(404)</span>
<span class="nc" id="L70">                .message(&quot;Родительской папки не существует&quot;)</span>
<span class="nc" id="L71">                .build());</span>
    }


}
</pre><div class="footer"><span class="right">Created with <a href="http://www.jacoco.org/jacoco">JaCoCo</a> 0.8.12.202403310830</span></div></body></html>